util/process_wrapper/system_posix.cc

```diff
@@ -174,12 +175,28 @@ int System::Exec(const System::StrType &executable,
     }
   }
 
-  int err, exit_status = -1;
+  int err, exit_status;
   do {
     err = waitpid(child_pid, &exit_status, 0);
   } while (err == -1 && errno == EINTR);
 
-  return exit_status;
+  if (WIFEXITED(exit_status)) {
+    return WEXITSTATUS(exit_status);
+  } else if (WIFSIGNALED(exit_status)) {
+    raise(WTERMSIG(exit_status));
+#ifdef WCOREDUMP
+  } else if (WCOREDUMP(exit_status)) {
+    std::cerr << "process wrapper error: child process produced a core dump.\n";
+    return -1;
+#endif
```

Comment:
I don't think this is quite right, given [the man page](https://man7.org/linux/man-pages/man2/wait.2.html):
>WCOREDUMP(wstatus)
              returns true if the child produced a core dump (see
              core(5)).  This macro should be employed only if
              WIFSIGNALED returned true.

Reply:
This seems like it's just additional information, but not strictly necessary.  Can you either incorporated it into the condition on line 185, or remove it?

Reply:
Good catch, removed.